crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -38,10 +38,10 @@ enum StateTransition<T: Component + Clone + Eq> {
 
 #[derive(Debug)]
 enum ScheduledOperation<T: Component + Clone + Eq> {
```

Comment:
It might be worth breaking apart the state and the `rerun` bool.
e.g.
```rust
enum ScheduledOperationType<T> {
    Set(T), Replace(T), Pop, Push(T),
}

struct ScheduledOperation<T> {
    operator_type: ScheduledOperationType<T>,
    rerun: ReRun,
}
```
especially since each variant has the `rerun` flag associated with it.
It would prevent having to write 
```rust
if !rerun {
    state.end_next_loop = true;
}
```
in each branch inside `state_cleaner`